lib/hawkular/operations/operations_api.rb

```diff
@@ -33,9 +33,31 @@ class Client < Hawkular::BaseClient
     # helper for parsing the "OperationName=json_payload" messages
     class WebSocket::Frame::Data
       def to_msg_hash
-        chunks = split('=', 2)
+        operation_name, json = split('=', 2)
 
-        { operationName: chunks[0], data: JSON.parse(chunks[1]) }
+        # Check if there is a zip file following JSON.
+        # This check is done only in the first 100KB, hoping it's unlikely to
+        # have such large amount of JSON before a zip file is attached.
+        magic_bits = ["\x50\x4B\x03\x04", "\x50\x4B\x05\x06", "\x50\x4B\x07\x08"]
+        search_chunk = json[0, 102_400]
```

Comment:
Hmm.... It can be done only in "live". It fails with VCR because it's mocking the `message` and there is no way to retrieve if it's binary or text.
So, I think I'll leave it as is for now.